app/comparator.py

```
@@ -0,0 +1,134 @@
+        Args:
+            repo_path: Path to the repository
+            old_commit: The older commit hash
+            new_commit: The newer commit hash
+            repo_name: Name of the repository
+            
+        Returns:
+            ComparisonResult containing diff content
+        """
+        try:
+            # Run the comparison script
+            result = subprocess.run(
```

Ruff (0.8.2)

`subprocess` call: check for execution of untrusted input

(S603)